### PR TITLE
feat: add BLS `sign` function.

### LINKED ISSUE #128: Implement BLS `sign` function
We want to just use an existing implementation for now.

I suggest that we create a `bls_nif` with `rustler` and we use the same Rust lib that lighthouse is using for now.

Here is the `bls` spec test for reference: https://github.com/sigp/lighthouse/blob/stable/testing/ef_tests/src/cases/bls_sign_msg.rs

Acceptance criteria:`sign` test case on bls spec should pass: https://github.com/lambdaclass/lambda_ethereum_consensus/blob/main/test/spec/runners/bls.ex#L63-L65

### PR DESCRIPTION
**Pull request by mpaulucci**
**Description**
Add bls `sign` function by using the lighthouse implementation

Closes #128

### CODE REVIEW

File: test/spec/runners/bls.ex [hunk 1]
```diff
@@ -60,8 +60,18 @@ defmodule BLSTestRunner do
...
+    %{message: message, privkey: private_key} = input
+
+    case output do
+      nil ->
+        {result, _error_msg} = Bls.sign(private_key, message)
+        assert result == :error
+
+      output ->
+        {:ok, signature} = Bls.sign(private_key, message)
+        assert signature == output
```
Comment (MegaRedHand) [inline test/spec/runners/bls.ex:73]: Prepending `assert` on match assignments improves error messages.
```suggestion
      nil ->
        assert {result, _error_msg} = Bls.sign(private_key, message)
        assert result == :error

      output ->
        assert {:ok, signature} = Bls.sign(private_key, message)
        assert signature == output
```

File: native/bls_nif/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    binary.into()
+}
+
+#[rustler::nif]
+fn sign<'env>(
+    env: Env<'env>,
+    private_key: Binary,
+    message: Binary,
+) -> Result<Binary<'env>, String> {
+    let sk = match SecretKey::deserialize(&private_key.as_slice()) {
```
Comment (MegaRedHand) [inline native/bls_nif/src/lib.rs:20]: Nit
```suggestion
    let sk = match SecretKey::deserialize(private_key.as_slice()) {
```

**Review by MegaRedHand — APPROVED** [review]

**Comment by MegaRedHand** [comment]
We should also add the crate to the `fmt` target in the _Makefile_

File: lib/bls.ex [hunk 3]
```diff
@@ -0,0 +1,27 @@
+defmodule Bls do
+  @moduledoc """
+  BLS signature verification.
+  """
+  use Rustler, otp_app: :lambda_ethereum_consensus, crate: "bls_nif"
+
+  @spec sign(binary(), binary()) :: binary()
```
[the remark this answers is not in the document]
Comment (mpaulucci) [reply lib/bls.ex:7]: good catch

**Pull request merged**